packages/p2p/src/components/modal-manager/modal-form/__tests__/modal-form.spec.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import { useStores } from 'Stores/index';
+import ModalForm from '../modal-form';
+
+const modal_form_props = {
+    initialValues: {
+        n: 0,
+    },
+    onSubmit: jest.fn(),
+};
+
+let mock_store: DeepPartial<ReturnType<typeof useStores>>;
+
+jest.mock('Stores', () => ({
+    ...jest.requireActual('Stores'),
+    useStores: jest.fn(() => mock_store),
+}));
+
+describe('<ModalForm />', () => {
+    beforeEach(() => {
+        mock_store = {
+            general_store: {
+                formik_ref: null,
+                saved_form_state: null,
+            },
+        };
+
+        mock_store.general_store.setFormikRef = (formik_ref: any) => {
```

Comment:
```suggestion
        mock_store.general_store.setFormikRef = (formik_ref: useRef<FormikProps<FormValues>>) => {
```

Reply:
https://stackoverflow.com/questions/66145167/object-is-possibly-null-typescript-react-useref-formik-innerref